ligand/views.py

```diff
@@ -172,164 +971,145 @@ def CachedTargetDetailsCompact(request, **kwargs):
+                    data_parsed[protein][record.source][assay] = {}
+                if record.value_type not in data_parsed[protein][record.source][assay].keys():
+                    if record.source == 'Guide to Pharmacology':
+                        data_parsed[protein][record.source][assay][record.value_type] = [x for x in record.p_activity_ranges.split('|')]
+                    else:
+                        data_parsed[protein][record.source][assay][record.value_type] = [record.p_activity_value]
                 else:
-                    # "No image available" SVG (source: https://commons.wikimedia.org/wiki/File:No_image_available.svg)
-                    picture = "https://upload.wikimedia.org/wikipedia/commons/thumb/a/ac/No_image_available.svg/600px-No_image_available.svg.png?20190827162820"
-
-                purchasability = vendors_dict[lig.id] if lig.id in vendors_dict.keys() else 0
-
-                data_parsed = {}
-                for record in records:
-                    assay = assay_conversion[record.assay_type]
-                    if record.source not in data_parsed.keys():
-                        data_parsed[record.source] = {}
-                    if assay not in data_parsed[record.source].keys():
-                        data_parsed[record.source][assay] = {}
-                    if record.value_type not in data_parsed[record.source][assay].keys():
-                        if record.source == 'Guide to Pharmacology':
-                            data_parsed[record.source][assay][record.value_type] = [x for x in record.p_activity_ranges.split('|')]
-                        else:
-                            data_parsed[record.source][assay][record.value_type] = [record.p_activity_value]
+                    if record.source == 'Guide to Pharmacology':
+                        data = [x for x in record.p_activity_ranges.split('|')]
+                        data_parsed[protein][record.source][assay][record.value_type] += data
                     else:
-                        if record.source == 'Guide to Pharmacology':
-                            data = [x for x in record.p_activity_ranges.split('|')]
-                            data_parsed[record.source][assay][record.value_type] += data
-                        else:
-                            data_parsed[record.source][assay][record.value_type].append(record.p_activity_value)
-
-                for source in data_parsed.keys():
-                    for assay_type in data_parsed[source].keys():
-                        for value_type in data_parsed[source][assay_type].keys():
-                            values = [float(x) for x in data_parsed[source][assay_type][value_type] if x != 'None']
+                        data_parsed[protein][record.source][assay][record.value_type].append(record.p_activity_value)
+            for protein in data_parsed.keys():
+            # for protein in sorted(list(data_parsed.keys())):
```

Comment:
please clean up

Reply:
Solved in [39bd4a8](https://github.com/protwis/protwis/pull/550/commits/39bd4a8230b189165eddc18d6beb1c8dbe956bd1)